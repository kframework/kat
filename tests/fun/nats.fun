// [ 1 : 2 : 3 : 4 : 5 : .Vals ]

let
    cons h [ t ] = [ h : t ]
 in
letrec
    map f  = fun [       ] -> [ ]
             |   [ h : t ] -> cons (f h) (map f [ t ])
and nats   = fun -> cons 1 (map (fun n -> n + 1) nats)
and take n = fun [       ] -> [ ]
             |   [ h : t ] -> if n > 0 then cons h (take (n - 1) [ t ]) else [ ]
 in take 5 [ 1 : 2 : 3 : 4 : 5 : 6 : 7 : .Exps ]
