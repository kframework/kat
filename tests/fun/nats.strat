f5bf8eb1522a640850678ed9961d3ba6 orig
# ad49e751a9a84bf057c0a2f7864fff01 exec
