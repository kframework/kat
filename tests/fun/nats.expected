
Strategy: # orig
================================================================================
SKIPPING

Strategy: # exec
================================================================================
SKIPPING
